### PR TITLE
Tag log source when print logs to console view for spark jobs

### PR DESCRIPTION
**Pull request by t-rufang**
- Logs when the  Spark job succeed
logsource
 - Logs when the Spark job failed
logsourcewitherror

### CODE REVIEW

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/ISparkBatchJob.java [hunk 1]
```diff
@@ -149,7 +154,43 @@
      * @return the job control message type and content pair observable
      */
     @NotNull
-    Observer<SimpleImmutableEntry<MessageInfoType, String>> getCtrlSubject();
+    Observer<SparkBatchJobLogLine> getCtrlSubject();
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/ISparkBatchJob.java:157]: There method can be introduced by a new interface `SparkClientControlMessage`

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/log/SparkBatchJobLogUtils.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static com.microsoft.azure.hdinsight.common.MessageInfoType.Error;
+import static com.microsoft.azure.hdinsight.common.MessageInfoType.*;
+
+public class SparkBatchJobLogUtils {
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/log/SparkBatchJobLogUtils.java:36]:
```suggestion
public class SparkLogUtils {
```

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/log/SparkLogSource.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.azure.hdinsight.spark.common.log;
+
+public class SparkLogSource {
+    public static final String Tool = "azuretool";
+    public static final String  Livy = "livy";
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/log/SparkLogSource.java:27]:
```suggestion
    public static final String Livy = "livy";
```

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/log/SparkLogSource.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.azure.hdinsight.spark.common.log;
+
+public class SparkLogSource {
+    public static final String Tool = "azuretool";
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/log/SparkLogSource.java:26]: These can be put into `SparkLogLine`

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/SparkDriverLogStreamReader.kt [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+import com.microsoft.azure.hdinsight.common.ConsoleViewLogLine
+import com.microsoft.azure.hdinsight.common.ConsoleViewTypeRegistration.Companion.contentTypeKeyMap
+import com.microsoft.azure.hdinsight.common.MessageInfoType
+import com.microsoft.azure.hdinsight.spark.common.log.SparkLogLine
+import com.microsoft.azure.hdinsight.spark.common.log.SparkLogUtils
+import java.io.InputStream
+import java.nio.charset.Charset
+import java.util.concurrent.Future
+
+class SparkDriverLogStreamReader(val processHandler: ProcessHandler,
```
[the remark this answers is not in the document]
Comment (t-rufang) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/SparkDriverLogStreamReader.kt:38]: There is a field `private val logType: Key<*>` in class `SparkSimpleLogStreamReader`. We don't need this field In class `SparkDriverLogStreamReader`. Instead, the `logType` need to be calculated based on the message each time we notify. 

---
In reply to: [413453003](https://github.com/microsoft/azure-tools-for-java/pull/4314#discussion_r413453003) [](ancestors = 413453003)

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/common/livy/interactive/Session.java [hunk 6]
```diff
@@ -207,7 +208,7 @@ public CreateParameters archives(final String... archives) {
         /**
          * Set Spark session configuration.
          *
-         * @param key key for Spark configuration to set into option
+         * @param key   key for Spark configuration to set into option
```
[the remark this answers is not in the document]
Comment (t-rufang) [reply Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/common/livy/interactive/Session.java:211]: The whitespace is introduced after I auto-format the code snippet in IntelliJ. The whitespace can make the description of the parameters align vertically.
image

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/common/livy/interactive/Session.java [hunk 7]
```diff
@@ -325,12 +329,12 @@ public PostSessions build() {
     /**
      * Create a Livy session instance.
      *
-     * @param name the session name which will be found in resource manager, such as Yarn
-     * @param baseUrl the connect URL of Livy, also the parent URL of submitting POST Livy session request,
-     *                ending with '/'
+     * @param name             the session name which will be found in resource manager, such as Yarn
```
[the remark this answers is not in the document]
Comment (t-rufang) [reply Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/common/livy/interactive/Session.java:332]: Same as the previous comment

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/log/SparkLogSource.java (hunk 3 above)
Comment (t-rufang) [reply Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/log/SparkLogSource.java:27]: Good catch, will fix this one~

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/log/SparkLogSource.java (hunk 4 above)
Comment (t-rufang) [reply Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/log/SparkLogSource.java:26]: Will fix this one

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/common/livy/interactive/Session.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (wezhang) [reply Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/common/livy/interactive/Session.java:211]: Looks Intellij rules have to be fixed.

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/log/SparkLogSource.java (hunk 3 above)
Comment (wezhang) [reply Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/log/SparkLogSource.java]: These static final string h

> Good catch, will fix this one~

Btw, these static final string names have to be upper.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/SparkDriverLogStreamReader.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (wezhang) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/SparkDriverLogStreamReader.kt:38]: Sure, let's keep this now.

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/log/SparkLogSource.java (hunk 3 above)
Comment (t-rufang) [reply Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/log/SparkLogSource.java]: Fixed in the latest commit

**Pull request merged**